internal/validation/strings.go

```diff
@@ -0,0 +1,12 @@
+package validation
+
+import (
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+var (
+	uuidRE       = regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$")
+	StringIsUuid = validation.StringMatch(uuidRE, "must be a valid UUID")
```

Comment:
The upstream [validation.IsUUID](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/validation/uuid.go) covers this need. The Parse function it relies on (from Hashicorp) works with upper, lower, and mixed case.

https://go.dev/play/p/U_z8v_5j2ip

Reply:
Good call, I like removing things!